homeassistant/components/egauge/coordinator.py

```diff
@@ -0,0 +1,92 @@
+"""Data update coordinator for eGauge energy monitors."""
+
+from __future__ import annotations
+
+from datetime import timedelta
+
+from egauge_async.exceptions import (
+    EgaugeAuthenticationError,
+    EgaugeException,
+    EgaugePermissionError,
+)
+from egauge_async.json.client import EgaugeJsonClient
+from egauge_async.json.models import RegisterInfo
+from httpx import ConnectError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_SSL,
+    CONF_USERNAME,
+    CONF_VERIFY_SSL,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryError
+from homeassistant.helpers.httpx_client import get_async_client
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import _LOGGER, DOMAIN
+from .models import EgaugeData
+
+type EgaugeConfigEntry = ConfigEntry[EgaugeDataCoordinator]
+
+
+class EgaugeDataCoordinator(DataUpdateCoordinator[EgaugeData]):
+    """Class to manage fetching eGauge data."""
+
+    serial_number: str
+    hostname: str
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        config_entry: ConfigEntry,
+    ) -> None:
+        """Initialize the coordinator."""
+        super().__init__(
+            hass,
+            logger=_LOGGER,
+            name=DOMAIN,
+            update_interval=timedelta(seconds=30),
+            config_entry=config_entry,
+        )
+        self.client = EgaugeJsonClient(
+            host=config_entry.data[CONF_HOST],
+            username=config_entry.data[CONF_USERNAME],
+            password=config_entry.data[CONF_PASSWORD],
+            client=get_async_client(
+                hass, verify_ssl=config_entry.data[CONF_VERIFY_SSL]
+            ),
+            use_ssl=config_entry.data[CONF_SSL],
+        )
+        # Populated in _async_setup
+        self._register_info: dict[str, RegisterInfo] = {}
+
+    async def _async_setup(self) -> None:
+        try:
+            self.serial_number = await self.client.get_device_serial_number()
+            self.hostname = await self.client.get_hostname()
+            self._register_info = await self.client.get_register_info()
+        except (EgaugeAuthenticationError, EgaugePermissionError) as err:
+            # will raise ConfigEntryAuthFailed once reauth is implemented
+            raise ConfigEntryError from err
+        except (ConnectError, EgaugeException) as err:
+            raise ConfigEntryError from err
```

Comment:
```suggestion
            raise UpdateFailed from err
```
or why wouldn't we retry here, so seem intermittent?